extensions/typescript-language-features/src/typescriptServiceClient.ts

```diff
@@ -1038,15 +1039,20 @@ export default class TypeScriptServiceClient extends Disposable implements IType
 				return;
 
 			case EventName.createDirectoryWatcher: {
-				const path = (event.body as Proto.CreateDirectoryWatcherEventBody).path;
-				if (path.startsWith(inMemoryResourcePrefix)) {
+				const fpath = (event.body as Proto.CreateDirectoryWatcherEventBody).path;
+				if (fpath.startsWith(inMemoryResourcePrefix)) {
+					return;
+				}
+				if (process.platform === 'darwin' && fpath === path.join(homedir(), 'Library')) {
```

Comment:
Just curious: what is the gist as to why TS watches `~/Library` in the end? Is that coming from a TS plugin?

Reply:
The dir watchers are created for ~/Library explicitly, because of a hard-coded ~/Library/caches in the typescript codebase.

The watcher comes from the ts plugin, yes. It happens when you create an in-memory typescript document, which also uses typing information which ends up accessing ~/Library/Caches/typescript. When resolving the in-memory doc to a projectRoot, the ts plugin decides the correct answer is ~/Library

Reply:
So I assume once this lands we can remove my change in https://github.com/microsoft/vscode/commit/6e8e1758574766a7c67cc22a0ced1c104347fac3 that I was not very happy about anyway because it will hide bugs like this one.

Reply:
Yes I think that's reasonable.